src/data/Data.js

```diff
@@ -147,7 +147,11 @@ Data.prototype = {
 
 		// Remove the expando if there's no more data
 		if ( key === undefined || jQuery.isEmptyObject( cache ) ) {
-			delete owner[ this.expando ];
+			// Webkit & Blink performance suffers when deleting properties
```

Comment:
I never tested safari myself so I'm not sure, I was just basing that on the comment: https://github.com/jquery/jquery/pull/1664#issuecomment-56916618